File: app/1_twosum.go
package app

func twoSum(nums []int, target int) []int {
	mymap := make(map[int]int)
	//create a dictionary to store key and values
	for i := 0; i < len(nums); i++ {
		j, ok := mymap[target-nums[i]]
		if ok {
			result := []int{j, i}
			return result
		}
		mymap[nums[i]] = i
	}
	result := []int{-1, 1}
	return result
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	twoSum(nums, target)
}

File: app/1_twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        d = {}  # assert the dictionary type to store value and key
        for i, num in enumerate(nums):
            if target-num in d:
                return [d[target-num], i]
            d[num] = i


nums = [2, 7, 11, 15]
target = 18
s = Solution()
print(s.twoSum(nums, target))
